File: 594-longest-harmonious-subsequence/main.go
func findLHS(nums []int) int {
    sort.Slice(nums, func(i, j int) bool {
        return nums[i] < nums[j]
    })

    maxLenght := 0
    j := 0

    for i := range nums {
        for nums[i] - nums[j] > 1 {
            j++
        }

        if nums[i] - nums[j] == 1 {
            maxLenght = max(maxLenght, i - j + 1)
        }
    }

    return maxLenght
}


func max(x, y int) int {
    if x < y {
        return y
    }
    return x
}

File: 594-longest-harmonious-subsequence/main.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        nums.sort()
        j = 0
        max_length = 0

        for i in range(len(nums)):
            while nums[i] - nums[j] > 1:
                j += 1
            
            if nums[i] - nums[j] == 1:
                max_length = max(max_length, i - j + 1)
        return max_length
